src/Shared/FileSystem/WindowsFileSystem.cs

```diff
@@ -22,50 +22,43 @@ internal enum FileArtifactType : byte
     }
 
     /// <summary>
-    /// Windows-specific implementation of file system operations using Windows native invocations
+    /// Windows-specific implementation of file system operations using Windows native invocations.
+    /// TODO For potential extra perf gains, provide native implementations for all IFileSystem methods and stop inheriting from ManagedFileSystem
     /// </summary>
-    internal class WindowsFileSystem : IFileSystem
+    internal class WindowsFileSystem : ManagedFileSystem
```

Comment:
nit: Should this class be sealed?

Reply:
`WindowsFileSystem` extends it to override some implementations with native ones.